integration_tests/src/main/python/csv_test.py

```diff
@@ -540,3 +540,13 @@ def test_csv_read_case_insensitivity(spark_tmp_path):
         lambda spark: spark.read.option('header', True).csv(data_path).select('one', 'two', 'three'),
         {'spark.sql.caseSensitive': 'false'}
     )
+
+@allow_non_gpu('FileSourceScanExec', 'CollectLimitExec', 'DeserializeToObjectExec')
+@pytest.mark.parametrize('data_gen', csv_supported_gens, ids=idfn)
+def test_csv_read_count(spark_tmp_path, data_gen):
```

Comment:
This is actually 15 cases. We can generate one CSV file with 15 columns in one case to save time.